### PR TITLE
Fix formatDateTime() with fractional negative datetime64

### LINKED ISSUE #33704: formatDatetime support for DateTime64 and fractions formatting

### PR DESCRIPTION
**Pull request by kardymonds**
Fixed https://github.com/ClickHouse/ClickHouse/issues/33704#issuecomment-1598704796 / https://github.com/ClickHouse/ClickHouse/issues/51262
Code same as [WriteHelpers.h](https://github.com/ClickHouse/ClickHouse/blob/master/src/IO/WriteHelpers.h#L785)

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix formatDateTime() if fractional negative datetime64.

### CODE REVIEW

File: src/Functions/formatDateTime.cpp [hunk 1]
```diff
@@ -943,7 +943,15 @@ class FunctionFormatDateTimeImpl : public IFunction
         {
             if constexpr (std::is_same_v<DataType, DataTypeDateTime64>)
             {
-                const auto c = DecimalUtils::split(vec[i], scale);
```
Comment (kardymonds) [inline src/Functions/formatDateTime.cpp:946]: It looks like for datetime64 needs a separate split function returning a second/fraction.

**Comment by den-crane** [comment]
Place new test into separate sql - `00718_format_datetime_1.sql`
It helps to track which test cases flakiness and ease reverting.

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit 65e5d40cae52b6cf9a2f0408f6ccb373f23a0e42 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51290/65e5d40cae52b6cf9a2f0408f6ccb373f23a0e42/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Functions/formatDateTime.cpp [hunk 2]
```diff
@@ -943,7 +943,15 @@ class FunctionFormatDateTimeImpl : public IFunction
...
+                auto c = DecimalUtils::split(vec[i], scale);
+
+                if (vec[i].value < 0 && c.fractional)
```
Comment (pufit) [inline src/Functions/formatDateTime.cpp:948]: Probably it also good to add a comment about this behavior

**Review by pufit — APPROVED** [review]

File: src/Functions/formatDateTime.cpp (hunk 2 above)
Comment (kardymonds) [reply src/Functions/formatDateTime.cpp:949]: Added (added also comment to DecimalUtils::split() in [other pull request](https://github.com/ClickHouse/ClickHouse/pull/51291/files#diff-176a35d6aab5cab74176aa902f1519abcb359e38e73da9eb0d5c9b40703d0840L50))

**Pull request merged**